src/config.ts

```diff
@@ -33,20 +43,48 @@ const defaultConfig: Config = {
     artifactsDir: 'build/contracts',
   },
 
-  blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
-  gasPrice: 20e9, // 20 gigawei
-
-  setupProvider: async baseProvider => baseProvider,
+  setupProvider: async (baseProvider) => baseProvider,
 
   coverage: false,
+
+  node: {
+    allowUnlimitedContractSize: false,
+    gasLimit: DEFAULT_BLOCK_GAS_LIMIT,
+    gasPrice: 20e9, // 20 gigawei
+  },
 };
 
 function getConfig(): Config {
   const location = findUp.sync(CONFIG_FILE, { type: 'file' });
   const providedConfig: Partial<Config> = location !== undefined && fs.existsSync(location) ? require(location) : {};
 
+  if (providedConfig.blockGasLimit !== undefined) {
+    warn(`blockGasLimit is deprecated. Use node.gasLimit instead. See ${configHelpUrl} for details.`);
+  }
+
+  if (providedConfig.gasPrice !== undefined) {
+    warn(`Please move gasPrice option inside node option. See ${configHelpUrl} for more details.`);
+  }
+
+  if (!!providedConfig.gasPrice && !!providedConfig.node?.gasPrice) {
```

Comment:
Unfortunately there's an issue here because `gasPrice` can be zero which is considered falsy. We should be using the `in` operator or comparing against `undefined`. (I would love it if TypeScript detected this error...)

I think comparing against `undefined` is better because it probably has better IDE support (because the `in` operator is used with a string).

```suggestion
  if (providedConfig.gasPrice !== undefined && providedConfig.node?.gasPrice !== undefined) {
```

Reply:
I agree that we should be explicit. On a sidenote - does 0 `gasPrice` makes any sense?

Reply:
We actually use gas price 0 to test some things in contracts because it simplifies some of the math.